src/Umbraco.Web.UI.Client/src/common/services/editor.service.js

```diff
@@ -635,6 +635,35 @@ When building a custom infinite editor view you can use the same components as a
       open(editor);
     }
 
+    /**
+     * @ngdoc method
+     * @name umbraco.services.editorService#contentTypeEditor
+     * @methodOf umbraco.services.editorService
+     *
+     * @description
+     * Opens the content type editor in infinite editing, the submit callback returns the alias of the saved content type.
+     *
+     * @param {object} editor rendering options.
+     * @param {number} editor.id Indicates the ID of the content type to be edited. Alternatively the ID may be set to `-1` in combination with `create` being set to `true` to open the content type editor for creating a new content type.
+     * @param {boolean} editor.create Set to `true` to open the content type editor for creating a new content type.
+     * @param {function} editor.submit Callback function when the submit button is clicked. Returns the editor model object.
+     * @param {function} editor.close Callback function when the close button is clicked.
+     * @returns {object} editor object.
+     */
+    function contentTypeEditor(editor, type) {
```

Comment:
Alternatively it could have just `editor` parameter, but `editor.type` or `editor.entityType` argument.

Reply:
We may also need to consider this for `contentTypePicker` which is currently for document types.

But we could probably pass in e.g. nodeType or entityType, e.g. documentType, mediaType or memberType and fallback to documentType as it is current behaviour. Maybe we should have documentTypePicker as well if developers want to use the specific type and we can easy between these.